tslearn/utils.py

```diff
@@ -28,6 +28,7 @@ def check_dims(X_fit, X):
         dimensions, except the first, does not match.
 
     """
+    # TODO: What is difference check_equal_size <--> this?
```

Comment:
I see 2 main differences:
* `check_equal_size` checks for actual size (ie. with trailing NaNs removed)
* `check_equal_size` does not check dimensionality (3rd component)